### PR TITLE
Update examples/java-agent plugin versions

### PR DESCRIPTION
**Pull request by briandealwis**
Update plugin versions for `examples/java-agent`, and tweak the example to listen on `$PORT` if provided, as provided by Cloud Run.

### CODE REVIEW

File: examples/java-agent/src/main/java/example/HelloWorld.java [hunk 1]
```diff
@@ -17,9 +17,14 @@
 package example;
 
 import static spark.Spark.get;
+import static spark.Spark.port;
```
Comment (loosebazooka) [inline examples/java-agent/src/main/java/example/HelloWorld.java]: The static import seems rather non-google style.
Should we be using `Spark.port(...)`, `Spark.get(...)`

File: examples/java-agent/pom.xml [hunk 2]
```diff
@@ -77,7 +77,9 @@
           <to>
             <image>gcr.io/REPLACE-WITH-YOUR-GCP-PROJECT/image-built-with-jib</image>
           </to>
-          <extraDirectory>${agent-extraction-root}</extraDirectory>
+          <extraDirectories>
+            <paths>${agent-extraction-root}</paths>
+          </extraDirectories>
           <container>
             <ports>4567</ports>
```
Comment (loosebazooka) [inline examples/java-agent/pom.xml]: does this need to be modified for cloud run $PORT?

File: examples/java-agent/src/main/java/example/HelloWorld.java (hunk 1 above)
Comment (briandealwis) [reply examples/java-agent/src/main/java/example/HelloWorld.java]: It is SparkJava style though.  For example: http://sparkjava.com and http://sparkjava.com/documentation#port

File: examples/java-agent/pom.xml (hunk 2 above)
Comment (briandealwis) [reply examples/java-agent/pom.xml]: We can't actually change this on the fly though.  The default for SparkJava is to listen on port 4567.  Cloud Run passes in a PORT and the container is required to listen on that port; Cloud Run doesn't care about the container image's exposed ports.

Let me add a note to the source and here to this effect.  I also forgot to update the build.gradle to use the `extraDirectories`.

File: examples/java-agent/src/main/java/example/HelloWorld.java [hunk 3]
```diff
@@ -17,9 +17,16 @@
...
 
 public class HelloWorld {
   public static void main(String[] args) {
-    get("/hello", (req, res) -> "Hello World");
+    // Allow use with Cloud Run which requires listening on the value in PORT
+    String portEnv = System.getenv("PORT");
```
Comment (chanseokoh) [inline examples/java-agent/src/main/java/example/HelloWorld.java]: Is it defined as 4567 by default somewhere? It should continue to work locally.

Comment (briandealwis) [reply examples/java-agent/src/main/java/example/HelloWorld.java]: SparkJava listens on 4567 by default (mentioned in the README.md)

File: examples/java-agent/src/main/java/example/HelloWorld.java [hunk 4]
```diff
@@ -17,9 +17,16 @@
...
+    if(portEnv != null) {
```
Comment (chanseokoh) [inline examples/java-agent/src/main/java/example/HelloWorld.java]: `if(portEnv` --> `if (portEnv`, or it's just not in Google style?

Comment (briandealwis) [reply examples/java-agent/src/main/java/example/HelloWorld.java]: Huh, I thought we had a validator to break the build.

Comment (TadCordle) [reply examples/java-agent/src/main/java/example/HelloWorld.java]: I don't think that runs on the example projects.

**Pull request merged**